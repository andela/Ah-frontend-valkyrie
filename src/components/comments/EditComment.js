import React, { Component, Fragment } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { editComment } from "./actions/actions";

export class EditComment extends Component {
  constructor(props) {
    super(props);
    this.state = {
      updatedCommentBody: "",
    };
  }

  handleOnChange = (event) => {
    this.setState({
      updatedCommentBody: event.target.value,
    });
  };

  handleSubmit = (event) => {
    event.preventDefault();
    const { articleSlug, commentId, commentBody } = this.props;
    const { updatedCommentBody } = this.state;
    const comment = {
      comment: {
        body: updatedCommentBody !== null ? updatedCommentBody : commentBody,
      },
    };
    this.props.editComment(articleSlug, commentId, comment);
  };

  render() {
    const { toggleForm, commentBody } = this.props;
    return (
      <Fragment>
        <form onSubmit={this.handleSubmit} className={toggleForm || "d-none"}>
          <div className="form-group">
            <textarea
              id="hide-text-box"
              name="body"
              defaultValue={commentBody}
              className="form-control expand-text-box"
              onChange={this.handleOnChange}
              required
            />
          </div>
          <div className="button">
            <input type="submit" value="Edit comment" className="btn btn-sm btn-primary mb-6" />
          </div>
        </form>
      </Fragment>
    );
  }
}

EditComment.propTypes = {
  articleSlug: PropTypes.string.isRequired,
  commentId: PropTypes.number.isRequired,
  toggleForm: PropTypes.string.isRequired,
  commentBody: PropTypes.string.isRequired,
  editComment: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  error: state.error,
});

export default connect(
  mapStateToProps,
  { editComment },
)(EditComment);
