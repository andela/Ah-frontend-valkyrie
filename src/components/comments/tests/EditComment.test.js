import React from "react";
import { shallow } from "enzyme";
import { EditComment } from "../EditComment";

describe("<Edit comment component />", () => {
  let mockComponent;
  const props = {
    articleSlug: "new-article-slug",
    commentId: 1,
    commentBody: "",
    editComment: jest.fn(),
  };
  const event = {
    target: {
      value: "change value",
    },
  };
  beforeEach(() => {
    mockComponent = shallow(<EditComment {...props} />);
  });
  it("renders as expected", () => {
    expect(mockComponent).toMatchSnapshot();
  });
  it("handle onChange", () => {
    mockComponent.instance().setState = jest.fn();

    mockComponent.instance().handleOnChange(event);

    expect(mockComponent.instance().setState).toBeCalled();
  });

  it("handle submit", () => {
    const event = {
      preventDefault: jest.fn(),
    };

    mockComponent.instance().handleSubmit(event);
    expect(props.editComment).toBeCalled();
  });
});
