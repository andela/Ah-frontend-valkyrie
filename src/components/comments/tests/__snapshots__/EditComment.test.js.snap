// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Edit comment component /> renders as expected 1`] = `ShallowWrapper {}`;
